packages/shopify-api/lib/auth/oauth/oauth.ts

```diff
@@ -178,19 +179,34 @@ export function callback(config: ConfigInterface): OAuthCallback {
       code: query.get('code'),
     };
 
-    const postParams = {
-      path: '/admin/oauth/access_token',
-      type: DataType.JSON,
-      data: body,
-    };
     const cleanShop = sanitizeShop(config)(query.get('shop')!, true)!;
 
-    const HttpClient = httpClientClass(config);
-    const client = new HttpClient({domain: cleanShop});
-    const postResponse = await client.post(postParams);
+    const postResponse = await abstractFetch(
+      `https://${cleanShop}/admin/oauth/access_token`,
+      {
+        method: 'POST',
+        body: JSON.stringify(body),
+        headers: {
+          'Content-Type': DataType.JSON,
+          Accept: DataType.JSON,
+        },
+      },
+    );
+
+    if (!postResponse.ok) {
+      throwFailedRequest(
```

Comment:
This maintains the old behaviour of throwing one of the `Http` errors if this request fails. I don't think any apps care about this, but it's non-breaking.